src/goto-instrument/contracts/doc/user/contracts-loops.md

```diff
@@ -168,6 +168,40 @@ void array_wipe(__CPROVER_size_t len, char * array)
 }
 ```
 
+### Caution With Nested Loop
+Due to the nature of @ref contracts-assigns, we need to be aware of the non-deterministic value of the assigned target.
+In the example below, 
+```c
+const unsigned table[256] = {1, 2, 3, ..., 256};
+
+void nested_loop_example() {
+        unsigned t = 0;
+        for( j=0; j<18; j++ )
+        __CPROVER_assigns(t, j, k) // t and k need to be included in the outer loop
+        __CPROVER_loop_invariant(0 <= j && j <= 18 && t < 256) // t < 256 must be included in the outer loop invariant
```

Comment:
Without the invariant clause `t < 256`, the program state for the inductive step allow `t` have arbitrary value and invalidate the inner loops' loop invariants.

Reply:
Reflected in [22b1aa6](https://github.com/diffblue/cbmc/pull/8377/commits/22b1aa6d81fbee491b383242f11e9dc451ecae8d)